File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int jumps=0,l=0,r=0;
        while(r<nums.length-1){
            int far = 0;
            for(int ind = l;ind <= r;ind++){
                far = Math.max(ind+nums[ind],far);
            }
            l=r+1;
            r=far;
            jumps++;
        }
        return jumps;
    }
}

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        jump = 0
        l = r = 0
        farthest = 0
        while(r < n-1):
            for i in range(l,r+1):
                farthest = max(farthest,i+nums[i])
            l = r+1
            r = farthest
            jump += 1
        return jump